raiden/network/rpc/smartcontract_proxy.py

```diff
@@ -116,11 +115,14 @@ def transact(
 
         slot = self.rpc_client.get_next_transaction()
         try:
+            to = None
+            if self.contract_address and len(self.contract_address) == 20:
+                to = self.contract_address
+            if to is None and self.contract.address and len(self.contract.address) == 20:
+                to = self.contract.address
+            assert to is not None
```

Comment:
This is a desperate attempt in working around inconsistent usage of `contract_address` vs `contract_address`. For some reason in some instances `contract.address` is becoming a `checksum` encoded hex address.
In some other instances `contract_address` is not initialized at all (while `contract.address` is and it is byte encoding).

Reply:
I guess this will go away with https://github.com/raiden-network/raiden/pull/5879 , but that will also require a maybe-not-so-nice rebase.